haystack/preview/dataclasses/answer.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass
+from haystack.preview.dataclasses.document import Document
+
+
+@dataclass(frozen=True)
+class Answer:
+    data: Any
+    question: str
+    metadata: Dict[str, Any]
+
+
+@dataclass(frozen=True)
+class ExtractiveAnswer(Answer):
```

Comment:
I just noticed that we could also call both classes `ExtractedAnswer` and `GeneratedAnswer`. I am not sure which one is better.